File: exercicios/java/Exercicio05/src/main/java/org/example/UsuarioBU.java
package org.example;

public abstract class UsuarioBU {
    protected long cpf;
    protected int diasDeEmprestimo;

    public UsuarioBU(long cpf, int diasDeEmprestimo) {
        this.cpf = cpf;
        this.diasDeEmprestimo = diasDeEmprestimo;
    }

    public long getCpf() {
        return cpf;
    }
    public void setCpf(long cpf) {
        this.cpf = cpf;
    }
    public int getDiasDeEmprestimo() {
        return diasDeEmprestimo;
    }
    public void setDiasDeEmprestimo(int diasDeEmprestimo) {
        this.diasDeEmprestimo = diasDeEmprestimo;
    }

    public String emprestar(String tituloDoLivro) {
        return "\" pegou emprestado o livro: " + tituloDoLivro + " com ";
    }
    public String devolver(String tituloDoLivro) {
        return "\" devolveu o livro: " + tituloDoLivro;
    }
}

File: exercicios/python/Exercicio05/usuario_bu.py
from abc import ABC, abstractmethod
class UsuarioBU(ABC):

    @abstractmethod
    def __init__(self, cpf:int, dias_de_emprestimo:int):
        self.__cpf = cpf
        self.__dias_de_emprestimo = dias_de_emprestimo
    
    @property
    def cpf(self):
        return self.__cpf
    
    @cpf.setter
    def cpf(self, cpf):
        self.__cpf = cpf
    
    @property
    def dias_de_emprestimo(self):
        return self.__dias_de_emprestimo
    
    @dias_de_emprestimo.setter
    def dias_de_emprestimo(self, dias_de_emprestimo):
        self.__dias_de_emprestimo = dias_de_emprestimo

    def emprestar (self, titulo_do_livro:str):
        return '" pegou emprestado o livro: ' + titulo_do_livro + ' com '

    def devolver (self, titulo_do_livro:str):
        return '" devolveu o livro: ' + titulo_do_livro